src/main/java/org/opentripplanner/middleware/utils/ItineraryUtils.java

```diff
@@ -0,0 +1,295 @@
+
+        // make sure itineraries have same amount of legs
+        if (previousItinerary.legs.size() != newItinerary.legs.size()) return false;
+
+        // make sure each leg matches
+        for (int i = 0; i < previousItinerary.legs.size(); i++) {
+            Leg previousItineraryLeg = previousItinerary.legs.get(i);
+            Leg newItineraryLeg = newItinerary.legs.get(i);
+
+            // for now don't analyze non-transit legs
+            if (!previousItineraryLeg.transitLeg) continue;
+
+            // make sure the same from/to stop are being used
+            if (
+                !stopsMatch(previousItineraryLeg.from, newItineraryLeg.from) ||
+                    !stopsMatch(previousItineraryLeg.to, newItineraryLeg.to)
+            ) {
+                return false;
+            }
+
+            // make sure the transit service is the same as perceived by the customer
+            if (
+                !equalsOrPreviousWasNull(previousItineraryLeg.mode, newItineraryLeg.mode) ||
+                    !equalsIgnoreCaseOrPreviousWasEmpty(previousItineraryLeg.agencyName, newItineraryLeg.agencyName) ||
+                    !equalsIgnoreCaseOrPreviousWasEmpty(
+                        previousItineraryLeg.routeLongName,
+                        newItineraryLeg.routeLongName
+                    ) ||
+                    !equalsIgnoreCaseOrPreviousWasEmpty(
+                        previousItineraryLeg.routeShortName,
+                        newItineraryLeg.routeShortName
+                    ) ||
+                    !equalsIgnoreCaseOrPreviousWasEmpty(
+                        previousItineraryLeg.headsign,
+                        newItineraryLeg.headsign
+                    ) ||
+                    !equalsOrPreviousWasNull(
+                        previousItineraryLeg.interlineWithPreviousLeg,
+                        newItineraryLeg.interlineWithPreviousLeg
+                    )
+            ) {
+                return false;
+            }
+
+            // make sure the transit trips are scheduled for the same time of the day
+            if (
+                !timeOfDayMatches(
+                    previousItineraryLeg.getScheduledStartTime(),
+                    newItineraryLeg.getScheduledStartTime()
+                ) || !timeOfDayMatches(
+                    previousItineraryLeg.getScheduledEndTime(),
+                    newItineraryLeg.getScheduledEndTime()
+                )
+            ) {
+                return false;
+            }
+        }
+
+        // if this point is reached, the itineraries are assumed to match
+        return true;
+    }
+
+    /**
+     * Checks whether two stops (OTP Places) match for the purposes of matching itineraries
+     */
+    private static boolean stopsMatch(Place stopA, Place stopB) {
+        // stop names must match
+        if (!StringUtils.equalsIgnoreCase(stopA.name, stopB.name)) return false;
```

Comment:
How come a third party lib is used to compare strings?

Reply:
Used String.equalsIgnoreCase in 52d438aad1badb194c212a736a16d72aff3da526